pages/components/buildcharactercomponents/cardcomponents/languagecomponents/Languageproperty.tsx

```diff
@@ -22,40 +28,73 @@ export default function Languageproperty(Props: PropsInterface) {
   const property = keysForNoDialect[0];
 
   const [selected, setSelect] = useState(() => {
-    if (typeof window !== "undefined") {
-      if (typeof dialectLanguage === "string") {
-        const saved: any = localStorage.getItem(
-          language + property + dialectLanguage
-        );
+    if (typeof dialectLanguage === "string") {
+      const value = language + property + dialectLanguage;
+      const thisLangProp = store.find((element: languageInterface) => {
+        if (element.value === value) {
+          return element;
+        } else {
+          return;
+        }
+      });
 
-        const intialValue = JSON.parse(saved);
-        return intialValue || false;
-      } else {
-        const saved: any = localStorage.getItem(language + property);
+      if (thisLangProp) {
+        return thisLangProp.selected;
+      }
+    } else {
+      const value = language + property;
+
+      const thisLangProp = store.find((element: languageInterface) => {
+        if (element.value === value) {
+          return element;
+        } else {
+          return;
+        }
+      });
 
-        const intialValue = JSON.parse(saved);
-        return intialValue || false;
+      if (thisLangProp) {
+        return thisLangProp.selected;
       }
     }
   });
   useEffect(() => {
     if (typeof dialectLanguage === "string") {
-      localStorage.setItem(
-        language + property + dialectLanguage,
-        JSON.stringify(selected)
-      );
       const value = language + property + dialectLanguage;
-      const languageProperty = { value, selected: selected };
-      return;
-      // dispatch(addLanguagePro(languageProperty));
+      if (store === undefined) {
+        return;
+      }
+      const thisLangProp = store.find((element: languageInterface) => {
+        if (element.value === value) {
+          return element;
+        } else {
+          return;
+        }
+      });
+
+      if (thisLangProp === undefined) {
```

Comment:
Nested if statement can become a little confusing - would concider how this flows - can some of the logic be stripped out into smaller fuctions?